packages/wrapper/src/editorAppClassic.ts

```diff
@@ -72,6 +77,20 @@ export class EditorAppClassic extends EditorAppBase {
                 monaco.editor.setTheme(languageDef.theme.name);
             }
         }
+
+        if (this.config.editorOptions?.['semanticHighlighting.enabled'] !== undefined) {
+            let userConfig;
+            try {
+                userConfig = JSON.parse(await getUserConfiguration());
+            } catch (e) {
+                userConfig = {};
+            }
+            const json = JSON.stringify({
+                ...userConfig,
+                'editor.semanticHighlighting.enabled': this.config.editorOptions['semanticHighlighting.enabled']
+            });
+            await updateUserConfiguration(json);
```

Comment:
I'm not sure to understand why this is needed?

btw, you still reformat the file, while using `configurationService.updateValue` would not

Reply:
> I'm not sure to understand why this is needed?

If you set it via the editor options https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/src/langium/langium-dsl/config/classicConfig.ts#L39-L43 the semanticHighlighting option is not applied. That's the reason why I brought back the workaround.

> btw, you still reformat the file, while using configurationService.updateValue would not

I don't understand what you mean

Reply:
You're parsing the json file, add a field then serialize it (and without formatting in single line)

VSCode includes method to update the user configuration without breaking the formatting of the configuration file (it just update or add the new field keeping the formatting of the rest of the file)

Reply:
Ok, now I understand. I just pushed an update.